ocaml/xapi/xapi_vm_snapshot.ml

```diff
@@ -57,115 +57,16 @@ let is_flag_set ~xs ~flag ~domid ~vm =
-    let snapid = List.sort compare_snapid_chunks snapid in
-    let read_chunk x = xs.Xenstore.Xs.read (snapshot_path ~xs ~domid ("snapid/" ^ x)) in
-    let snapid = String.concat "" (List.map read_chunk snapid) in
-
-    (* Get the uuid of the snapshot VM *)
-    debug "wait_for_snapshot: getting uuid of the snapshot VM";
-    let snapshot_uuid =
-      try xs.Xenstore.Xs.read (snapshot_path ~xs ~domid "snapuuid")
-      with _ ->
-        error "The snapshot has not been correctly created; did snapwatchd create a full VM snapshot?";
-        raise (Api_errors.Server_error (Api_errors.vm_snapshot_with_quiesce_failed, [ Ref.string_of vm ])) in
-    let snapshot_ref =
-      try Db.VM.get_by_uuid ~__context ~uuid:snapshot_uuid
-      with _ ->
-        error "The snapshot UUID provided by snapwatchd is not a valid UUID.";
-        raise (Api_errors.Server_error (Api_errors.vm_snapshot_with_quiesce_failed, [ Ref.string_of vm ])) in
-
-    Db.VM.set_transportable_snapshot_id ~__context ~self:snapshot_ref ~value:snapid;
-    Db.VM.set_name_label ~__context ~self:snapshot_ref ~value:new_name;
-
-    (* update the snapshot-info field *)
-    Db.VM.remove_from_snapshot_info ~__context ~self:snapshot_ref ~key:Xapi_vm_clone.disk_snapshot_type;
-    Db.VM.add_to_snapshot_info ~__context ~self:snapshot_ref ~key:Xapi_vm_clone.disk_snapshot_type ~value:Xapi_vm_clone.quiesced;
-
-    (* Update is-vmss-snapshot field for snapshot taken from VMSS policy *)
-    if Xapi_vmss.is_vmss_snapshot ~__context then
-      Db.VM.set_is_vmss_snapshot ~__context ~self:snapshot_ref  ~value:true;
-
-    snapshot_ref
-
-  | "snapshot-error" ->
-    (* If an error was occured we get the error type and return *)
-    let error_str = xs.Xenstore.Xs.read (snapshot_path ~xs ~domid "error") in
-    let error_code () = try xs.Xenstore.Xs.read (snapshot_path ~xs ~domid "error/code") with _ -> "0" in
-    error "wait_for_snapshot: %s" error_str;
-    if List.mem error_str [
-        Api_errors.xen_vss_req_error_init_failed;
-        Api_errors.xen_vss_req_error_prov_not_loaded;
-        Api_errors.xen_vss_req_error_no_volumes_supported;
-        Api_errors.xen_vss_req_error_start_snapshot_set_failed;
-        Api_errors.xen_vss_req_error_adding_volume_to_snapset_failed;
-        Api_errors.xen_vss_req_error_preparing_writers;
-        Api_errors.xen_vss_req_error_creating_snapshot;
-        Api_errors.xen_vss_req_error_creating_snapshot_xml_string ]
-    then raise (Api_errors.Server_error (error_str, [ Ref.string_of vm; error_code () ]))
-    else raise (Api_errors.Server_error (Api_errors.vm_snapshot_with_quiesce_failed, [ Ref.string_of vm; error_str ]))
-
-  | e ->
-    failwith (Printf.sprintf "wait_for_snapshot: unexpected result (%s)" e)
-
-(* We fail if the guest does not support quiesce mode. Normally, that should be detected *)
-(* dynamically by the xapi_vm_lifecycle.update_allowed_operations call.                  *)
 let snapshot_with_quiesce ~__context ~vm ~new_name =
```

Comment:
I think you can delete the entire function definition, as it is no longer called from anywhere?